conans/test/functional/editable/layouts_test.py

```diff
@@ -0,0 +1,83 @@
+# coding=utf-8
+
+import os
+import re
+import textwrap
+import unittest
+
+from conans.test.utils.tools import TestClient
+from conans.util.files import load, save_files
+from conans.client.cache import LAYOUTS_FOLDER
+
+
+class LayoutTest(unittest.TestCase):
+
+    def test_missing_layouts(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+            class Pkg(ConanFile):
+                pass
+            """)
+
+        client.save({"conanfile.py": conanfile})
+        client.run("link . mytool/0.1@user/testing -l=missing")
+        client2 = TestClient(client.base_folder)
+        consumer = textwrap.dedent("""
+            [requires]
+            mytool/0.1@user/testing
+            """)
+        client2.save({"conanfile.txt": consumer})
+        client2.run("install .", assert_error=True)
+        self.assertIn("ERROR: Layout file not found: missing", client2.out)
```

Comment:
What if exists at link time, but then it is removed when you actually need it? Or you can link and add it later the layout file. Also, it might find the repo file, but then you remove it, and then it will find a cache layout one. This check is totally necessary, irrespective if other check could be introduced at link time, but I think that one gives almost no value.

Reply:
I agree with @danimtb, I think that the check is very useful to avoid errors (misspellings, confusion resolving relative paths,...) at the moment of writing the `conan link` command. This is aligned with my comment about the `--layout` argument: https://github.com/conan-io/conan/pull/4287#discussion_r247824870

Reply:
Added this check now ``conan link --layout`` will raise if the layout file is not found or incorrect.